invokeai/app/invocations/baseinvocation.py

```diff
@@ -470,14 +472,32 @@ class BaseInvocation(ABC, BaseModel):
 
     @classmethod
     def get_all_subclasses(cls):
+        app_config = InvokeAIAppConfig.get_config()
+        app_config.parse_args()
```

Comment:
```suggestion
```

As per https://github.com/invoke-ai/InvokeAI/pull/3466: "In non-main scripts, it is not necessary (or recommended) to call parse_args()" - it also causes issues with unrecognized CLI args, such as `uvicorn` args used in commercial dev. I verified this is working without `parse_args()`.

Reply:
It does not work without parse_args unless parse_args is called before this function is called in graph.py. The alternative is to call parse_args in api_app before and of the api stuff is imported, but this seems easier to break

At any rate, we are calling parse_args in api_app anyways, so I can't imagine it's a major issue?

Reply:
Ok, I see how this won't work for commercial.

The problem is when the Graph model is created, we need `InvokeAIAppConfig` to be fully loaded - eg, we need `parse_args()` to have been called once. Then, when the union of valid nodes is constructed, they are filtered out by the config.

When you tested and It *appeared* to work without calling `parse_args()`, it only hid the nodes from the UI. That it hid the nodes was an accidental side-effect related to how the openapi schema is massaged for use by the UI. The backend still happily executes the denied nodes.

If `parse_args()` is not allowed due to commercial's set up (FWIW, it *is* called in OSS `api_app.py`), we need to figure out how to ensure the config file has been read by the time we grab the config inside `baseinvocation.py`.

Reply:
to be clear, this *does* work for commercial without `parse_args()`

Reply:
Reviewed commercial and understand how it *does* actually work, cool.

I've update the PR to work for OSS without needing to call `parse_args()` in `baseinvocation.py` - by rearranging the imports in `api_app.py` and `cli_app.py`.